src/usecases/services/ReadService.ts

```diff
@@ -28,14 +28,14 @@ export class ReadService<TModel, TSearchDto = ISearchInputDto> {
      * Model class
      * @protected
      */
-    protected modelClass;
+    protected modelClass: any;
```

Comment:
Не знаю насколько такое предложение применимо, но может ограничим тип объектом? Понятно что число или булевое сюда вряд ли будут, но всё-таки:
```suggestion
    protected modelClass: object;
```

Reply:
any там используется осознанно, т.к. тип object в гарде приводит к ошибке ts: The right-hand side of an instanceof expression must be of type any or of a type assignable to the Function interface type.

Как вариант могу предложить использовать IType (наша копия интерфейса Type из @nestjs/common), но тогда в generic'е необходимо указать, что TModel extends IType (что в общем-то является правдой).

Reply:
> что TModel extends IType

Тогда Typescript будет ругаться при экстенде от TModel. Тут наверно можно просто сделать `protected modelClass: IType<TModel>`
![image](https://github.com/user-attachments/assets/d1436c87-de1d-41f6-a16b-38e0dbda3cd1)